src/Features/Lsif/Generator/Program.cs

```diff
@@ -284,6 +285,10 @@ private static async Task LocateAndRegisterMSBuild(TextWriter logFile, Directory
                 var generationStopwatch = Stopwatch.StartNew();
                 await lsifGenerator.GenerateForProjectAsync(project, GeneratorOptions.Default, cancellationToken);
                 await logFile.WriteLineAsync($"Generation for {project.FilePath} completed in {generationStopwatch.Elapsed.ToDisplayString()}.");
+
+                // Remove the project from the workspace; we reuse the same workspace object to ensure that some workspace-level services (like the IMetadataService
+                // or IDocumentationProviderService) are kept around allowing their caches to be reused.
```

Comment:
> IDocumentationProviderService) are kept around allowing their caches to be reused.

Seems like this would be good to have right above `new AdHocWorkspace()` also/instead?

Reply:
One place is fine, and it's not like the method is large.